source/ConditionsStore.cpp

```diff
@@ -417,8 +418,26 @@ ConditionsStore::DerivedProvider &ConditionsStore::GetProviderPrefixed(const str
 	auto it = providers.emplace(std::piecewise_construct,
 		std::forward_as_tuple(prefix),
 		std::forward_as_tuple(prefix, true));
-	storage[prefix].provider = &(it.first->second);
-	return it.first->second;
+	DerivedProvider *provider = &(it.first->second);
+	if(VerifyProviderLocation(prefix))
+	{
+		storage[prefix].provider = provider;
+		// Check if any matching later entries within the prefixed range use the same provider.
+		auto checkIt = storage.find(prefix);
+		while(checkIt != storage.end() && (0 == checkIt->first.compare(0, prefix.length(), prefix)))
+		{
+			const string &ceName = checkIt->first;
+			ConditionEntry &ce = checkIt->second;
+			if(ce.provider != provider)
+			{
+				Logger::LogError("Error: replacing condition-entry for \"" + ceName + "\", because it is within range of prefixed derived provider \"" + prefix + "\".");
```

Comment:
```suggestion
				Logger::LogError("Error: replacing condition-entry for \"" + ceName + "\""
						", because it is within range of prefixed derived provider \"" + prefix + "\".");
```